src/config.c

```diff
@@ -0,0 +1,197 @@
+        rm_config.cf_bucket_size.max = new_val / 2 < 255 ? new_val / 2 : 255;
+    }
+
+    return REDISMODULE_OK;
+}
+static RedisModuleString *getNumericValue(const char *name, void *privdata) {
+    return RedisModule_CreateStringPrintf(NULL, "%zu", *((size_t *)privdata));
+}
+
+typedef struct {
+    const char *name;
+    const char *defVal;
+    RedisModuleConfigSetStringFunc setValue;
+    RedisModuleConfigGetStringFunc getValue;
+    void *privdata;
+} RM_ConfigVar;
+
+RM_ConfigVar rm_config_vars[] = {
+    {
+        .name = "bf-error-rate",
+        .defVal = "0.01",
+        .setValue = setFloatValue,
+        .getValue = getFloatValue,
+        .privdata = &rm_config.bf_error_rate,
+    },
+    {
+        .name = "bf-initial-size",
+        .defVal = "100",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.bf_initial_size,
+    },
+    {
+        .name = "bf-expansion-factor",
+        .defVal = "2",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.bf_expansion_factor,
+    },
+    {
+        .name = "cf-bucket-size",
+        .defVal = "2",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_bucket_size,
+    },
+    {
+        .name = "cf-initial-size",
+        .defVal = "1024",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_initial_size,
+    },
+    {
+        .name = "cf-max-iterations",
+        .defVal = "20",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_max_iterations,
+    },
+    {
+        .name = "cf-expansion-factor",
+        .defVal = "1",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_expansion_factor,
+    },
+    {
+        .name = "cf-max-expansions",
+        .defVal = "32",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_max_expansions,
+    },
+};
+
+int RegisterStringConfig(RedisModuleCtx *ctx, RM_ConfigVar var) {
+    return RedisModule_RegisterStringConfig(ctx, var.name, var.defVal,
+                                            REDISMODULE_CONFIG_UNPREFIXED, var.getValue,
+                                            var.setValue, NULL, var.privdata);
+}
+
+int RM_RegisterConfigs(RedisModuleCtx *ctx) {
+    RedisModule_Log(ctx, "notice", "Registering configuration options");
+
+    for (int i = 0; i < sizeof rm_config_vars / sizeof *rm_config_vars; ++i) {
+        RM_ConfigVar var = rm_config_vars[i];
+        if (RegisterStringConfig(ctx, var) != REDISMODULE_OK) {
+            RedisModule_Log(ctx, "warning", "Failed to register config option `%s`", var.name);
+            return REDISMODULE_ERR;
+        }
+    }
+
+    return RedisModule_LoadConfigs(ctx);
```

Comment:
Should this function really have two responsibilities? Why can't we simply rename it then, at the very least, to `RegisterAndLoadConfiguration`?